Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Delete.ts

```diff
@@ -7,26 +7,40 @@ import { DeleteResult, Logger, DeleteRequest } from '@edfi/meadowlark-core';
 import type { PoolClient, QueryResult } from 'pg';
 import {
   deleteDocumentByIdSql,
-  checkIsReferencedDocumentSql,
   deleteReferencesSql,
   referencedByDocumentSql,
+  deleteExistenceIdsByDocumentId,
+  existenceIdsForDocument,
+  existenceIdsToVerify,
 } from './SqlHelper';
 
 export async function deleteDocumentById(
   { id, validate, traceId }: DeleteRequest,
   client: PoolClient,
 ): Promise<DeleteResult> {
   let deleteResult: DeleteResult = { response: 'UNKNOWN_FAILURE' };
+  let references;
 
   try {
     client.query('BEGIN');
 
     if (validate) {
-      // Check for any references to the document to be deleted
-      const referencesResult = await client.query(checkIsReferencedDocumentSql(id));
+      // Check for any references to the document to be deleted (including itself)
+      const existenceIdResult = await client.query(existenceIdsForDocument(id));
+      // If the record doesn't exist, exit
+      if (!existenceIdResult || existenceIdResult.rowCount === 0) {
```

Comment:
I'm uncomfortable with `!existenceIdResult`. The type is `any`, so I have to mentally parse `existenceIdResult.rowCount === 0` to conclude it's `object | null | undefined`. Easier to just use `existenceIdResult != null`.

Reply:
Or the whole if could be `existenceIdResult?.rowCount === 0`

Reply:
Updated to use existenceIdResult?.rowCount === 0